setup.py

```diff
@@ -57,17 +57,6 @@
     'setuptools',
     ]
 
-# typing_extensions is needed with Python 3.7 and older, types imported
-# from that package (Literal, Protocol, TypedDict, ...) become available
-# from typing since Python 3.8.
-install_requires.append("typing-extensions>=3.7.4.3; python_version < '3.8'")
```

Comment:
We will need `typing-extensions` for HW requirements, with different Python requirement, `python_version < '3.10'`. Discovered yesterday...

Reply:
> We will need `typing-extensions` for HW requirements, with different Python requirement, `python_version < '3.10'`. Discovered yesterday...

3.10? What is in 3.10? I only found `Self` missing in '<3.11', and that should not be required by install anyway, no?

Reply:
`TypeAlias`, I'm afraid. But, yeah, it might be needed just for type-checking time, not in runtime.

Reply:
maybe I'm blind, but I don't see it being used in codebase. In some opened PR perhaps? :)

> needed just for type-checking time, not in runtime.

In #2209, I'm adding typing_extensions only to "dev virtualenv", not as a hard dependency. Make sense, no?

Reply:
> maybe I'm blind, but I don't see it being used in the codebase. In some opened PR perhaps? :)

Yep, https://github.com/teemtee/tmt/pull/1316

> > needed just for type-checking time, not in runtime.
> 
> In #2209, I'm adding typing_extensions only to "dev virtualenv", not as a hard dependency. Make sense, no?

Yes, that should work out.

No idea how to proceed. I guess it's fine dropping it here after all, I'm afraid I can't make more changes to my PR anyway, not until the dust settles and all heads of this Python 3.6 deprecation hydra are resolved and merged :/

Reply:
Hm, should I keep the typing-extensions then? I'd prefer to drop it now to have clean plate and restore with your PR as it adds that requirement again.

Reply:
> all heads of this Python 3.6 deprecation hydra are resolved

You make the best comments @happz :smile:

Reply:
> Hm, should I keep the typing-extensions then?

I think the consensus is they should not be in install_requires anyway. Make sense to have clean slate as you wrote.

Reply:
Yes, drop it, I'll add it if/as needed in my patch (probably it's going to be needed, but last night I got an idea that might help us get rid of it in the long run).